src/psyclone/psyad/tl2ad.py

```diff
@@ -292,7 +292,7 @@ def _add_precision_symbol(symbol, table):
         kind_symbol = symbol.copy()
         kind_symbol.interface = ImportInterface(kind_contr_sym)
         table.add(kind_symbol)
-    elif symbol.is_automatic or symbol.is_modulevar or symbol.is_constant:
+    elif symbol.is_automatic or symbol.is_modulevar or symbol.is_static:
```

Comment:
The `symbol` here is a precision symbol and this code is really checking that we have a declaration for it because if we don't know where it has come from we can't safely generate the adjoint code. Having written that I've realised there's a nicer way of checking for this condition so I've changed the code. Hopefully it's better now.